CMakeLists.txt

```diff
@@ -269,6 +270,10 @@ if (QL_BUILD_TEST_SUITE OR QL_BUILD_BENCHMARK)
     add_subdirectory(test-suite)
 endif()
 
+if (QL_BUILD_FUZZ_TEST_SUITE AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    add_subdirectory(fuzz-test-suite)
+endif()
```

Comment:
If `QL_BUILD_FUZZ_TEST_SUITE` is set for an unsupported compiler then CMake configuration should fail. Also, is this feature supported with clang-cl?

Reply:
Fair enough. I admit my cmake foo is somewhat lacking. I'll see what I can work out but if you have suggestions on how you'd like to do that then I'd appreciate your guidance 🙂

Reply:
I'm not familiar with clang-cl so I'd have to look into that a bit more

Reply:
Something like the following should be good enough:

```cmake
if (QL_BUILD_FUZZ_TEST_SUITE)
    if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
        add_subdirectory(fuzz-test-suite)
    else()
        message(FATAL_ERROR "QL_BUILD_FUZZ_TEST_SUITE is currently only supported for Clang")
    endif()
endif()
```

Reply:
Sorry I didn't see this and I ended up using [cmake_dependent_option](https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html) instead. Let me know if that works otherwise, I'll put this in instead as it looks like it achieves roughly the same thing.

Reply:
Either way is fine to me but I’ll defer to what @lballabio prefers.

Reply:
@lballabio @pcaspers 
Did we abandon support for older Ubuntu versions than 22 (CMake 3.22 is available in Ubuntu 22 / Debian 11)? 

Or maybe we should not be restricted by system specific versions and let users use a later CMake version?  This may mean Opensourcerisk/Engine need do the same upgrade.

Reply:
I see.  No, I'd try relaxing the requirement then.  @silvergasp, what do you think?

Reply:
Yeah, honestly the change in version was only to allow for [cmake_dependent_option](https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html). While the ergonomics are a little more clunky we could revert back to the earlier version, this would just become a regular `option` and an if statement which is a fairly minor change. I don't feel strongly about going either way.

Worth noting however that I did make/test this change on Ubuntu 20.04, although I had forgotten that I was using the [pip version of cmake](https://pypi.org/project/cmake/), which backports a newer version to older systems.

Reply:
I'll open a PR soon, reverting to an earlier version of cmake.

Reply:
thanks!

Reply:
Rolled back here #1831